packages/subproviders/src/subproviders/metamask_subprovider.ts

```diff
@@ -0,0 +1,124 @@
+import { marshaller, Web3Wrapper } from '@0xproject/web3-wrapper';
+import { JSONRPCRequestPayload, Provider } from 'ethereum-types';
+import * as ethUtil from 'ethereumjs-util';
+
+import { Callback, ErrorCallback } from '../types';
+
+import { Subprovider } from './subprovider';
+
+/**
+ * This class implements the [web3-provider-engine](https://github.com/MetaMask/provider-engine)
+ * subprovider interface and the provider sendAsync interface.
+ * It handles inconsistencies with Metamask implementations of various JSON RPC methods.
+ * It forwards JSON RPC requests involving the domain of a signer (getAccounts,
+ * sendTransaction, signMessage etc...) to the provider instance supplied at instantiation. All other requests
+ * are passed onwards for subsequent subproviders to handle.
+ */
+export class MetamaskSubprovider extends Subprovider {
```

Comment:
To call `super.handleRequest` in the default of the switch statement?

Reply:
Well `MetamaskSubprovider` is a `SignerProvider` isn't it? Looking at it, the only benefit would be not needing to also instantiate the web3Wrapper instance variable here since we re-implemented all the same RPC requests.